File: pset6/credit.py
from cs50 import get_int

credit_int = 0
while credit_int < 1:
    credit_int = get_int("Number: ")

credit_str = str(credit_int)
size = len(credit_str)
sum_final = 0

for i in range(size - 2, -1, -2):
    soma = int(credit_str[i]) * 2
    sum_final += (int(soma / 10 % 10)) + (soma % 10)
    soma = 0

for i in range(size - 1, -1, -2):
    sum_final += int(credit_str[i])

begin = credit_str[0] + credit_str[1]
valid = (sum_final % 10) == 0

if (valid and size == 15 and int(begin) in [34, 37]):
    print("AMEX")
elif (valid and size == 16 and int(begin) in range(51, 55 + 1)):
    print("MASTERCARD")
elif (valid and size in [13, 16] and int(begin[0]) == 4):
    print("VISA")
else:
    print("INVALID")

File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    
    long number;
    int sum = 0;
    int sum_final = 0;
    long calc = 0;
    int begin = 0;
    
    
    //trava1
    do
    {
        number = get_long("Number: ");
    }
    while (number < 0);
    
    //checkvalid
    for (long divisor = 10; divisor < 10000000000000000; divisor = divisor * 100)
    {
        calc = number;
        calc = (calc / divisor) % 10;
        sum = calc * 2;
        sum_final = sum_final + (sum / 10 % 10) + (sum % 10);
        sum = 0;
        calc = 0;
    }
    for (long divisor = 1; divisor < 10000000000000000; divisor = divisor * 100)
    {
        calc = number;
        calc = (calc / divisor) % 10;
        sum_final = sum_final + calc;
        calc = 0;
    }
  
    
    //convert begin to int
    begin = (number / pow(10, 12));
    
    //which card
    if ((sum_final % 10) != 0)
    {
        printf("INVALID\n");
    }
    else if (((begin / 10) % 100) == 34 || ((begin / 10) % 100) == 37)
    {
        printf("AMEX\n");
    }
    else if (((begin / 100) % 100) >= 51 && ((begin / 100) % 100) <= 55)
    {
        printf("MASTERCARD\n");
    }    
    else if ((begin % 10) == 4 || ((begin / 1000) % 10) == 4)
    {
        printf("VISA\n");
    }        
    else
    {
        printf("INVALID\n");        
    }
}